src/kmer.jl

```diff
@@ -0,0 +1,31 @@
+# kmer.jl
+# =======
+#
+# Basic kmer types.
+#
+# This file is a part of the Kmers.jl, a package in the BioJulia ecosystem.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+abstract AbstractKmer
+
+for inttype in subtypes(Unsigned)
+    nbits = sizeof(inttype) * 8
+    typename = "Kmer$nbits"
+    @eval begin
+        bitstype $nbits $(parse(typename)){T<:NucleicAcid, K} <: AbstractKmer
```

Comment:
@bicycle1885 I'd probably like to keep 0.5.x compat, just until 0.6.0 lands. So long as there are not many places to change, and deprecation warnings will be triggered to indicate what needs to change. The first actual kmers.jl release will hopefully be 0.6.0 only.

(Rationale: I'm going to need to be doing some prototyping with this package pretty soon, and I want it to "just work" with 0.5.0, which is what I currently run. I'm trying to avoid always running nightlies as much as possible.)